api/sirsidynix_authentication_provider.py

```diff
@@ -0,0 +1,332 @@
+        # This suffix can be part of the blocked patron types list
+        patron_suffix = patron_type[len(self.sirsi_library_prefix) :]
+        if patron_suffix in self.sirsi_disallowed_prefixes:
+            patrondata.block_reason = SirsiBlockReasons.PATRON_BLOCKED
+            return patrondata
+
+        # Get patron "fines" information
+        status = self.api_patron_status_info(
+            patron_key=patrondata.permanent_id,
+            session_token=patrondata.session_token,
+        )
+
+        if not status or "fields" not in status:
+            return None
+
+        status_fields: dict = status["fields"]
+        fines = status_fields.get("estimatedFines")
+        if fines is not None:
+            # We ignore currency for now, and assume USD
+            patrondata.fines = float(fines.get("amount", 0))
+
+        # Blockable statuses
+        if status_fields.get("hasMaxDaysWithFines") or status_fields.get("hasMaxFines"):
+            patrondata.block_reason = PatronData.EXCESSIVE_FINES
+        elif status_fields.get("hasMaxLostItem"):
+            patrondata.block_reason = PatronData.TOO_MANY_LOST
+        elif status_fields.get("hasMaxOverdueDays") or status_fields.get(
+            "hasMaxOverdueItem"
+        ):
+            patrondata.block_reason = PatronData.TOO_MANY_OVERDUE
+        elif status_fields.get("hasMaxItemsCheckedOut"):
+            patrondata.block_reason = PatronData.TOO_MANY_LOANS
+        elif status_fields.get("expired"):
+            patrondata.block_reason = SirsiBlockReasons.EXPIRED
+
+        # If previously, the patron was blocked this should unset the value in the DB
+        if patrondata.block_reason is None:
+            patrondata.block_reason = PatronData.NO_VALUE
+        patrondata.complete = True
+        return patrondata
+
+    ###
+    # API requests
+    ###
+
+    def _request(
+        self, method: str, path: str, json=None, session_token=None
+    ) -> Response:
+        """Request wrapper that adds the relevant request headers.
+
+        :param method: The HTTP method for the request
+        :param path: The url path that will get joined to the server_url, should not have a leading '/'
+        :param json: The json data to be sent to the API endpoint
+        """
+        if path.startswith("/"):
+            raise ValueError(
+                f"Sirsidynix URL path {path} should not have a leading '/'"
+            )
+        headers = {
+            "SD-Originating-App-Id": self.sirsi_app_id,
+            "SD-Working-LibraryID": self.sirsi_library_id,
+            "x-sirs-clientID": self.sirsi_client_id,
+        }
+        if session_token:
+            headers["x-sirs-sessionToken"] = session_token
+
+        url = urljoin(self.server_url, path)
+        # Adding a long timeout because /patronStatusInfo would fail often
+        return HTTP.request_with_timeout(
+            method, url, headers=headers, json=json, timeout=120
```

Comment:
😟 that timeout. I think maybe we need to drop the timeout down... since a timeout this long will cause issues in production with other libraries.

Reply:
Removed the timeout. Today the API was snappy (kind of) but on other days I've had long waits for a response. Maybe production is better .